ambry-store/src/test/java/com.github.ambry.store/BlobStoreStatsTest.java

```diff
@@ -241,8 +315,351 @@ public void testValidDataSizeAfterDeletes() throws InterruptedException, StoreEx
+   * @throws StoreException
+   * @throws IOException
+   */
+  @Test
+  public void testBucketingCoverageTransition() throws InterruptedException, StoreException, IOException {
+    assumeTrue(bucketingEnabled);
+    CountDownLatch scanStartedLatch = new CountDownLatch(1);
+    // do not hold the initial scan
+    MockThrottler mockThrottler = new MockThrottler(scanStartedLatch, new CountDownLatch(0));
+    throttlers.put(BlobStoreStats.IO_SCHEDULER_JOB_TYPE, mockThrottler);
+    // add a put that is going to expire in 20 seconds
+    long initialScanTimeInMs = state.time.milliseconds();
+    BlobStoreStats blobStoreStats = setupBlobStoreStats(2, 0);
+    // proceed only when the scan is started
+    assertTrue("IndexScanner took too long to start", scanStartedLatch.await(5, TimeUnit.SECONDS));
+    verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(initialScanTimeInMs, 0L));
+    verifyAndGetContainerValidSize(blobStoreStats, initialScanTimeInMs);
+    // advance time by 20 seconds
+    state.advanceTime(TimeUnit.SECONDS.toMillis(20));
+    // hold the next scan
+    CountDownLatch scanHoldLatch = new CountDownLatch(1);
+    scanStartedLatch = new CountDownLatch(1);
+    mockThrottler.holdLatch = scanHoldLatch;
+    mockThrottler.startedLatch = scanStartedLatch;
+    mockThrottler.isThrottlerStarted = false;
+    assertTrue("IndexScanner took too long to start", scanStartedLatch.await(5, TimeUnit.SECONDS));
+    int throttleCountBeforeRequests = mockThrottler.throttleCount.get();
+    verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(initialScanTimeInMs, 0L));
+    verifyAndGetContainerValidSize(blobStoreStats, initialScanTimeInMs);
+    assertEquals("Throttle count shouldn't have changed", throttleCountBeforeRequests,
+        mockThrottler.throttleCount.get());
+    // request something outside of the forecast coverage
+    verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(0L, 0L));
+    verifyAndGetContainerValidSize(blobStoreStats, 0L);
+    scanHoldLatch.countDown();
+    blobStoreStats.close();
+  }
+
+  @Test
+  public void testBucketingWithLowIndexEntries() throws InterruptedException, StoreException, IOException {
+    state.destroy();
+    assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+    tempDir = StoreTestUtils.createTempDirectory("blobStoreStatsDir-" + UtilsTest.getRandomString(10));
+    state = new CuratedLogIndexState(isLogSegmented, tempDir, false, false);
+    int bucketCount = bucketingEnabled ? 1 : 0;
+    BlobStoreStats blobStoreStats = setupBlobStoreStats(bucketCount, 0);
+    verifyAndGetContainerValidSize(blobStoreStats, state.time.milliseconds());
+    verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(state.time.milliseconds(), 0));
+    state.addPutEntries(3, CuratedLogIndexState.PUT_RECORD_SIZE, Utils.Infinite_Time);
+    blobStoreStats = setupBlobStoreStats(bucketCount, 0);
```

Comment:
can you also add the test to cover the path where in we go from empty store to some more entries added to the store using same blobStoreStats instance